src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -162,94 +177,33 @@ public string ToString(bool withZeroX)
         public static bool operator <=(ValueKeccak left, ValueKeccak right) => left.CompareTo(right) <= 0;
     }
 
-    /// <summary>
-    /// Used as dictionary key with implicit conversion to devirtualize comparisions
-    /// </summary>
-    [DebuggerStepThrough]
-    public readonly struct KeccakKey : IEquatable<KeccakKey>, IComparable<KeccakKey>
-    {
-        public byte[] Bytes { get; }
-
-        private KeccakKey(byte[] bytes)
-        {
-            Bytes = bytes;
-        }
-
-        public static implicit operator KeccakKey(Keccak k) => new(k.Bytes);
-
-        public int CompareTo(KeccakKey other)
-        {
-            return Extensions.Bytes.Comparer.Compare(Bytes, other.Bytes);
-        }
-
-        public bool Equals(KeccakKey other)
-        {
-            if (ReferenceEquals(Bytes, other.Bytes))
-            {
-                return true;
-            }
-
-            if (Bytes is null)
-            {
-                return other.Bytes is null;
-            }
-
-            if (other.Bytes is null)
-            {
-                return false;
-            }
-
-            return Extensions.Bytes.AreEqual(Bytes, other.Bytes);
-        }
-
-        public override bool Equals(object? obj)
-        {
-            return obj is KeccakKey && Equals((KeccakKey)obj);
-        }
-
-        public override int GetHashCode()
-        {
-            if (Bytes is null) return 0;
-
-            long v0 = Unsafe.ReadUnaligned<long>(ref MemoryMarshal.GetArrayDataReference(Bytes));
-            long v1 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(Bytes), sizeof(long)));
-            long v2 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(Bytes), sizeof(long) * 2));
-            long v3 = Unsafe.ReadUnaligned<long>(ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(Bytes), sizeof(long) * 3));
-
-            v0 ^= v1;
-            v2 ^= v3;
-            v0 ^= v2;
-
-            return (int)v0 ^ (int)(v0 >> 32);
-        }
-    }
-
     [DebuggerStepThrough]
+    [DebuggerDisplay("{ToString()}")]
     public class Keccak : IEquatable<Keccak>, IComparable<Keccak>
     {
         public const int Size = 32;
 
         public const int MemorySize =
-            MemorySizes.SmallObjectOverhead +
+            MemorySizes.SmallObjectOverhead -
```

Comment:
Why does it become `-`?

Reply:
`SmallObjectOverhead` is `24`; but it only uses `16`

8 bytes object header, 8 bytes method table, no padding, 32 byte struct
```
Type layout for 'Keccak'
Size: 32 bytes. Paddings: 0 bytes (%0 of empty space)
|=========================================|
| Object Header (8 bytes)                 |
|-----------------------------------------|
| Method Table Ptr (8 bytes)              |
|=========================================|
|  0-31: ValueKeccak Bytes (32 bytes)     |
| |=====================================| |
| |  0-31: Vector256`1 Bytes (32 bytes) | |
| | |=============================|     | |
| | |   0-7: UInt64 _00 (8 bytes) |     | |
| | |-----------------------------|     | |
| | |  8-15: UInt64 _01 (8 bytes) |     | |
| | |-----------------------------|     | |
| | | 16-23: UInt64 _02 (8 bytes) |     | |
| | |-----------------------------|     | |
| | | 24-31: UInt64 _03 (8 bytes) |     | |
| | |=============================|     | |
| |=====================================| |
|=========================================|
```